### PR TITLE
Refactoring of .NET Core tests for CSLA 6

### PR DESCRIPTION
**Pull request by TheCakeMonster**
Fixes to tests for CSLA 6 on .NET Core 3.1

Forms part of unit test fixes detailed in issue #2544

I don't think the tests in DataPortalClientTests.cs are relevant anymore as a result of the DataPortalProxyFactory class having been retired; they are commented out for review. I think we've also done away with the ability to use BinaryFormatter, so there is a test in SerializationTests.cs that is also commented out, pending review.

### CODE REVIEW

File: Source/csla.netcore.test/Configuration/FluentConfigTests.cs [hunk 1]
```diff
@@ -22,159 +22,159 @@
...
-    public void FluentConfigSecurity()
-    {
-      new CslaConfiguration()
-        .Security().PrincipalCacheMaxCacheSize(123);
-
-      Assert.AreEqual(123, Csla.Security.PrincipalCache.MaxCacheSize, "MaxCacheSize");
-    }
-  }
+  //  [TestClass]
+  //  public class FluentConfigTests
```
Comment (rockfordlhotka) [inline Source/csla.netcore.test/Configuration/FluentConfigTests.cs:26]: I agree with removing this code.

File: Source/csla.netcore.test/DataPortal/DataPortalClientTests.cs [hunk 2]
```diff
@@ -27,86 +27,87 @@ public class DataPortalClientTests
...
-        { ProxyTypeName = typeof(Csla.DataPortalClient.HttpProxy).AssemblyQualifiedName, DataPortalUrl = "https://example.com/test" });
-      Csla.DataPortalClient.DataPortalProxyFactory.AddDescriptor("abc", fake);
-
-      var factory = new Csla.DataPortalClient.DataPortalProxyFactory();
-      var proxy = factory.Create(typeof(ResourceType));
-      Assert.IsNotNull(proxy, "proxy can't be null");
-      Assert.IsInstanceOfType(proxy, typeof(Csla.DataPortalClient.HttpProxy), "should be httpproxy");
-      Assert.AreEqual("https://example.com/test", ((Csla.DataPortalClient.HttpProxy)proxy).DataPortalUrl);
-    }
+    // I don't think these tests are relevant anymore, since the changes for CSLA 6 do away with DataPortalProxyFactory
```
Comment (rockfordlhotka) [inline Source/csla.netcore.test/DataPortal/DataPortalClientTests.cs:35]: Agreed

File: Source/csla.netcore.test/Serialization/SerializationTests.cs [hunk 3]
```diff
@@ -49,32 +52,37 @@ public void UseMobileFormatter()
...
-        Assert.AreEqual(ApplicationContext.SerializationFormatters.BinaryFormatter, serializer);
-        var s = Csla.Serialization.SerializationFormatterFactory.GetFormatter();
-        Assert.IsInstanceOfType(s, typeof(Csla.Serialization.BinaryFormatterWrapper));
-      }
-      finally
-      {
-        Csla.Configuration.ConfigurationManager.AppSettings["CslaSerializationFormatter"] = "MobileFormatter";
-      }
-    }
+    // BinaryFormatter dropped as of CSLA 6?
```
Comment (rockfordlhotka) [inline Source/csla.netcore.test/Serialization/SerializationTests.cs:55]: Correct, no more BinaryFormatter, so related tests can go away

**Comment by rockfordlhotka** [comment]
The test project(s) need to be added to the `csla.test.sln` solution to run as part of the CI build.

**Review by rockfordlhotka — APPROVED** [review]

**Comment by TheCakeMonster** [comment]
There are a couple of outstanding issues here:

1. The configuration tests are currently disabled - tests in ReadConfigFromDotnetCoreSubsystemTests.cs. Because they affect global state, they break all of the other tests, which can run in parallel with them. We could try to solve this using some form of locking, but that would add constant complexity to the test solution. However, moving them into another test project might be the best approach.
2. Two of the dashboard tests fail - those associated with counting of failures. This might indicate that there is a bug with the dashboard code in the area of exception handling within the dataportal, but I've not yet had a chance to investigate this. These tests are disabled in DashboardTests.cs
3. We've got a problem with the build. Some of the tests that use DI fail as soon as Csla.AspNetCore is built. I haven't managed to work out why, despite a day of trying. Deleting the netcoreapp3.1 folder from the Bin folder is the temporary workaround if this occurs - but that's not a solution!

**Pull request merged**